src/app/pages/reports/api-keys/api-keys.component.js

```diff
@@ -17,32 +19,63 @@ export const ReportsApiKeysComponent = {
         }
 
         $onInit () {
+            this.clearApiKeyFilter();
             this.search();
         }
 
+        clearApiKeyFilter () {
+            this.activityRange = {
+                range: 30,
+                startDate: new Date(),
+                endDate: new Date(),
+            };
+            this.activityRange.startDate.setDate(this.activityRange.endDate.getDate() - this.activityRange.range + 1); // offset to account for inclusion of endDate in range
+        }
+
+        createFilterDataObject () {
+            let filterData = {};
+            filterData.startDate = this.ReportService.coerceToMidnight(this.activityRange.startDate);
+            filterData.endDate = this.ReportService.coerceToMidnight(this.activityRange.endDate);
+            filterData.dateAscending = this.activityRange.dateAscending;
+            filterData.apiKeyFilter = this.filter;
+            return filterData;
+        }
+
         dateAdjust (obj) {
             var ret = angular.copy(obj);
             ret.startDate = this.ReportService.coerceToMidnight(ret.startDate);
             ret.endDate = this.ReportService.coerceToMidnight(ret.endDate, true);
             return ret;
         }
 
-        prepare (results) {
-            this.displayed = results.map(item => item);
-        }
-
         search () {
-            let that = this;
             this.networkService.getApiUserActivity(this.dateAdjust(this.activityRange))
                 .then(results => {
-                    that.results = results;
-                    that.prepare(that.results);
+                    this.results = results;
```

Comment:
```suggestion
                    this.results = results.map(item => {
                        item.friendlyCreationDate = this.$filter('date')(item.date, 'MMM d, y H:mm:ss');
                        return item;
                    });
                    // and remove the entire "this.displayed" section
```

I feel like we've talked about this, and I know you were going to change it around. I thought you did, but somehow it doesn't look like what I expected. Sorry, I probably approved it earlier without catching that

Reply:
Hmm, it looks strange in this code snippet. But, I added a this.results variable to accommodate `<div st-table="$ctrl.displayed" st-safe-src="$ctrl.results">`
For some reason, in the above code snippet, it looks like results is being assigned, and then right below being mutated with the results of the map. Anyway, this pattern is what I've seen in other controllers - providing the table with the necessary 'displayed' collection and original dataset.

Reply:
I was looking at the full file to make sure I saw it correctly. The problem is that `this.displayed` should _not_ be set in the controller/component. The smart-table directive handles the data flow from `this.results` to the `this.displayed` variable. In some of the other reports it's not done correctly, but it is in the search page. The `this.displayed` variable should not exist in the controller/component at all